core/arch/arm/sm/psci.c

```diff
@@ -149,9 +149,7 @@ void tee_psci_handler(struct thread_smc_args *args)
 			;
 		break;
 	case PSCI_SYSTEM_RESET:
-		psci_system_off();
-		while (1)
-			;
+		psci_system_reset();
```

Comment:
According to PSCI documentation, 5.1.10, SYSTEM_RESET, it does not return value.